### PR TITLE
Hw stream practice

### PR DESCRIPTION
**Pull request by OlehMytiai**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -1,80 +1,69 @@
...
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(n -> List.of(n.split(REGEX)))
+                .flatMap(Collection::stream)
+                .mapToInt(Integer::parseInt)
+                .boxed()
+                .filter(i -> i % 2 == 0)
+                .min(Comparator.comparingInt(a -> a))
```
Comment (DaniilOpryshko) [inline src/main/java/practice/StreamPractice.java:22]: We can simplify this
```suggestion
                .flatMap((e) -> Array.stream(e.split(REGEX)))
                .map(Integer::parseInt)
                .filter(i -> i % 2 == 0)
                .min(Integer::compareTo)
```

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -1,80 +1,69 @@
...
+    public static final String REGEX = ",";
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(n -> List.of(n.split(REGEX)))
+                .flatMap(Collection::stream)
+                .mapToInt(Integer::parseInt)
+                .boxed()
+                .filter(i -> i % 2 == 0)
```
Comment (DaniilOpryshko) [inline src/main/java/practice/StreamPractice.java:21]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api#use-validator-properly

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -1,80 +1,69 @@
...
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 == 1 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(integer -> integer % 2 == 1)
```
Comment (DaniilOpryshko) [inline src/main/java/practice/StreamPractice.java:30]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api#use-validator-properly

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -1,80 +1,69 @@
...
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == Person.Sex.MAN
+                        && p.getAge() >= fromAge
```
Comment (DaniilOpryshko) [inline src/main/java/practice/StreamPractice.java:38]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api#use-validator-properly

File: src/main/java/practice/StreamPractice.java [hunk 5]
```diff
@@ -1,80 +1,69 @@
...
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
```
Comment (DaniilOpryshko) [inline src/main/java/practice/StreamPractice.java:46]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api#use-validator-properly

File: src/main/java/practice/StreamPractice.java [hunk 6]
```diff
@@ -1,80 +1,69 @@
...
-    /**
-     * Given a List of `Person` instances (having `name`, `age`, `sex` and `cats` fields,
-     * and each `Cat` having a `name` and `age`),
-     * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
-     */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == Person.Sex.WOMAN
+                        && p.getAge() >= femaleAge)
```
Comment (DaniilOpryshko) [inline src/main/java/practice/StreamPractice.java:55]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api#use-validator-properly

**Review by DaniilOpryshko — CHANGES_REQUESTED** [review]

**Review by DaniilOpryshko — COMMENTED** [review]
GJ!

**Review by DaniilOpryshko — APPROVED** [review]

**Review by AlexRozv — APPROVED** [review]
great job 🔥

**Pull request closed without merging**